core/src/main/java/io/github/wysohn/triggerreactor/core/manager/trigger/TriggerInfo.java

```diff
@@ -66,18 +59,56 @@ public String getTriggerName() {
         return triggerName;
     }
 
-    public boolean isSync(){
+    public void put(TriggerConfigKey key, Object value) {
+        ValidationUtil.notNull(config);
+
+        config.put(key.getKey(), value);
+    }
+
+    public void put(TriggerConfigKey key, int index, Object value) {
+        ValidationUtil.notNull(config);
+
+        config.put(key.getKey(index), value);
+    }
+
+    public <T> Optional<T> get(TriggerConfigKey key, Class<T> clazz) {
+        Optional<T> old = Optional.ofNullable(config)
+                .flatMap(c -> c.get(key.getOldKey(), clazz));
+        if(old.isPresent())
+            return old;
```

Comment:
The old key should be updated with a valid key name. Each Trigger config creates a new key if no key is expected in this PR, which will cause confusion and nightmare for server devs. For instance:
```json
{
  "event": "org.spongepowered.api.event.entity.living.player.CooldownEvent$End",
  "Event": "org.spongepowered.api.event.entity.living.player.CooldownEvent$End",
  "//": "^ what? Which key would be used correctly? Have no idea ¯\_(ツ)_/¯"
}
```

Reply:
Yeah that was my concern too, but removing old one is also risky cause admittedly the key name of sync is 'sync' in Custom Trigger and 'Sync' in Command Trigger, so it might delete the legit config.

I think the mistake of 'deleting some config accidentally' have higher impact on the server than having some confusion for the devs. Well, kind of moving the responsibility to the devs, yet some triggers already have inconsistent keys (sync or Sync in CustomTrigger and CommandTrigger), so there is really no work around I assume.

I'm open for any suggestions though

Reply:
> but removing old one is also risky cause admittedly the key name of sync is 'sync' in Custom Trigger and 'Sync' in Command Trigger, so it might delete the legit config.

Oh yeah, that would be the hot point of this issue. Should be better to release a minor version at the same time as an official release, not a beta release including notify about this migration concern, I think.

Reply:
So we are sticking with this implementation but make sure to notify that there can be duplicated keys?

Reply:
Yeah, absolutely.